toolkit/offchain/src/csl.rs

```diff
@@ -88,6 +88,7 @@ pub(crate) fn get_builder_config(
 		.ref_script_coins_per_byte(&convert_reference_script_costs(
 			&protocol_parameters.min_fee_reference_scripts.clone(),
 		)?)
+		.deduplicate_explicit_ref_inputs_with_regular_inputs(true)
```

Comment:
Cons: allows balancing TXs that are constructed with "lower quality", when our code does thing sub-optimally.
Pros: it will be in place if we are in situation when not using this requires a lot of effort

I think it can stay